### PR TITLE
Speed up builds on macOS

### PR DESCRIPTION
**Pull request by Serock3**
Fixes DES-193.

An issue with the `--target` flag to cargo build caused caching not to work properly on macOS slowing down the CI.

- - -
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/5602)

### CODE REVIEW

**Review by raksooo — COMMENTED** [review]
This looks good! Before merging I would like to try building on both x86 and ARM and install both builds on both architectures :) Have you tried cross compiling for aarch64 on x86 Linux? `TARGETS=aarch64-unknown-linux-gnu` ./build.sh. You can try installing and running it on the raspberry pi on the test desk :)

> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/5602#-:-Nm5ptdmDWzkLvMVQtRv:b48qvpr)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @Serock3)

___
*[`build.sh` line 319 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/5602#-Nm5c08y5KVfxhsl470L:-Nm5c08y5KVfxhsl470M:b6248qx) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/a23a2de5e2473c06d802743d69fe07d5dd00ce85/build.sh#L319)):*
> ```Shell
> 
> EMPTY=()
> for t in "${TARGETS[@]:-"${EMPTY[@]}"}"; do
> ```

Could we make this work without specifying `EMPTY`? Would this give the same result?
```
for t in "${TARGETS[@]:-""}"; do
```

I think it would loop once with `""` as `t`. I'll let you decide what you prefer.

**Review by Pururun — COMMENTED** [review]
While running `./build-apk.sh --dev-build --no-docker` I get the following error on my Intel Mac:

```
Building mullvad-daemon for aarch64-linux-android
(...)

   **Compiling** mullvad-jni v0.0.0 (/Users/jonatanrhodin/checkout/mullvadvpn-app/mullvad-jni)

**error****:** could not compile \`mullvad-jni\` (lib)

  

Caused by:
(...)
  (signal: 11, SIGSEGV: invalid memory reference)
```

I can send you the full error if needed.

> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/5602#-:-Nn8GJJVDsPitnAf2RMH:b-rndeby)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @Serock3)

**Review by Serock3 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/5602)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @raksooo)

___
*[`build.sh` line 319 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/5602#-Nm5c08y5KVfxhsl470L:-Nn8W5z-Al-oPDXNcFEa:bijfnr4) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/a23a2de5e2473c06d802743d69fe07d5dd00ce85/build.sh#L319)):*
<details><summary><i>Previously, raksooo (Oskar Nyberg) wrote…</i></summary><blockquote>

Could we make this work without specifying `EMPTY`? Would this give the same result?
```
for t in "${TARGETS[@]:-""}"; do
```

I think it would loop once with `""` as `t`. I'll let you decide what you prefer.
</blockquote></details>

This seems to work the same, and is cleaner. I remember struggling to get this loop to work properly in all cases and this is so simple I wonder why I didn't figure it out myself... Maybe there's some edge case I have forgotten about. Until I find any issues I will use your implementation.

**Review by Serock3 — COMMENTED** [review]
This error was not reproducible. Possibly caused by an unrelated change.

I have now compiled and tested the app on aarch64 and x86, and it seems to be working as expected.

> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/5602#-:-Nn9ac4v0MkPEcNLP6ii:byrskm)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @raksooo)

**Review by raksooo — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/5602#-:-NnPE7fjCuvLRpad7T9T:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

___
*[`build.sh` line 319 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/5602#-Nm5c08y5KVfxhsl470L:-NnPDusN5OvwV4G1b90q:btojbfd) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/a23a2de5e2473c06d802743d69fe07d5dd00ce85/build.sh#L319)):*
<details><summary><i>Previously, Serock3 (Sebastian Holmin) wrote…</i></summary><blockquote>

This seems to work the same, and is cleaner. I remember struggling to get this loop to work properly in all cases and this is so simple I wonder why I didn't figure it out myself... Maybe there's some edge case I have forgotten about. Until I find any issues I will use your implementation.
</blockquote></details>

Great!

**Pull request merged**
